apps/token-manager/app/src/components/AppLayout.js

```diff
@@ -0,0 +1,106 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import styled from 'styled-components'
+import { AppBar, AppView, Button, ButtonIcon, Viewport, font } from '@aragon/ui'
+import MenuButton from './MenuButton/MenuButton'
+
+const AppLayout = ({
+  children,
+  title,
+  afterTitle,
+  smallViewPadding,
+  largeViewPadding,
+  onMenuOpen,
+  mainButton,
+}) => {
+  return (
+    <Viewport>
+      {({ below }) => (
+        <AppView
+          padding={below('medium') ? smallViewPadding : largeViewPadding}
+          appBar={
+            <AppBar>
+              <AppBarContainer
+                style={{ padding: below('medium') ? '0' : '0 30px' }}
+              >
+                <Title>
+                  {below('medium') && <MenuButton onClick={onMenuOpen} />}
+                  <TitleLabel>{title}</TitleLabel>
+                  {afterTitle}
+                </Title>
+                {mainButton &&
+                  (below('medium') ? (
+                    <ButtonIcon
+                      onClick={mainButton.onClick}
+                      title={mainButton.label}
+                      css={`
+                        width: auto;
+                        height: 100%;
+                        padding: 0 20px 0 10px;
+                        margin-left: 8px;
+                      `}
+                    >
+                      {mainButton.icon}
+                    </ButtonIcon>
+                  ) : (
+                    <Button mode="strong" onClick={mainButton.onClick}>
+                      {mainButton.label}
+                    </Button>
+                  ))}
+              </AppBarContainer>
+            </AppBar>
+          }
+        >
+          {children}
+        </AppView>
+      )}
+    </Viewport>
+  )
+}
+
+AppLayout.defaultProps = {
+  smallViewPadding: 20,
+  largeViewPadding: 30,
```

Comment:
Should this be `mediumViewPadding`?

Reply:
In the context of this component, there are only two modes: it doesn’t (and shouldn’t IMO) know about the global breakpoints. That’s why I was previously avoiding using “small” and “large”, and go for distinct names like “compact” instead.